src/triangle.rs

```diff
@@ -0,0 +1,343 @@
+        ),
+        (0.0, 0.0),
+        (1.0, 0.0),
+    );
+
+    /// A new [`Triangle`] from three vertices ([`Points`])
+    #[inline]
+    pub fn new(a: impl Into<Point>, b: impl Into<Point>, c: impl Into<Point>) -> Self {
+        Self {
+            a: a.into(),
+            b: b.into(),
+            c: c.into(),
+        }
+    }
+
+    /// A new [`Triangle`] from three float vertex coordinates
+    #[inline]
+    pub const fn from_coords(a: (f64, f64), b: (f64, f64), c: (f64, f64)) -> Self {
+        Self {
+            a: Point::new(a.0, a.1),
+            b: Point::new(b.0, b.1),
+            c: Point::new(c.0, c.1),
+        }
+    }
+
+    /// The centroid of the [`Triangle`]
+    #[inline]
+    pub fn centroid(&self) -> Point {
+        (1.0 / 3.0 * (self.a.to_vec2() + self.b.to_vec2() + self.c.to_vec2())).to_point()
+    }
+
+    /// The circumcenter of the [`Triangle`]
+    #[inline]
+    pub fn circumcenter(&self) -> Point {
+        let d = 2.0
+            * (self.a.x * (self.b.y - self.c.y)
+                + self.b.x * (self.c.y - self.a.y)
+                + self.c.x * (self.a.y - self.b.y));
+
+        let ux = ((self.a.x.powi(2) + self.a.y.powi(2)) * (self.b.y - self.c.y)
+            + (self.b.x.powi(2) + self.b.y.powi(2)) * (self.c.y - self.a.y)
+            + (self.c.x.powi(2) + self.c.y.powi(2)) * (self.a.y - self.b.y))
+            / d;
+
+        let uy = ((self.a.x.powi(2) + self.a.y.powi(2)) * (self.c.x - self.b.x)
+            + (self.b.x.powi(2) + self.b.y.powi(2)) * (self.a.x - self.c.x)
+            + (self.c.x.powi(2) + self.c.y.powi(2)) * (self.b.x - self.a.x))
+            / d;
+
+        Point::new(ux, uy)
+    }
+
+    /// The offset of each vertex from the centroid
+    #[inline]
+    pub fn offsets(&self) -> [Vec2; 3] {
+        let centroid = self.centroid().to_vec2();
+
+        [
+            (self.a.to_vec2() - centroid),
+            (self.b.to_vec2() - centroid),
+            (self.c.to_vec2() - centroid),
+        ]
+    }
+
+    /// The area of the [`Triangle`]
+    #[inline]
+    pub fn area(&self) -> f64 {
+        0.5 * (self.b - self.a).cross(self.c - self.a)
+    }
+
+    /// Whether this [`Triangle`] has zero area
+    #[doc(alias = "is_empty")]
+    #[inline]
+    pub fn is_zero_area(&self) -> bool {
+        self.area() == 0.0
+    }
+
+    /// Inradius of [`Triangle`] (the greatest radius of a circle that is within the [`Triangle`])
+    /// with center [`Triangle::circumcenter`]
+    #[inline]
+    pub fn inradius(&self) -> f64 {
+        let ab = self.a.distance(self.b);
+        let bc = self.b.distance(self.c);
+        let ac = self.a.distance(self.c);
+
+        2.0 * self.area() / (ab + bc + ac)
+    }
+
+    /// Circumradius of [`Triangle`] (the smallest radius of a circle such that it intercepts each vertex)
+    /// with center [`Triangle::circumcenter`]
+    #[inline]
+    pub fn circumradius(&self) -> f64 {
```

Comment:
Yeah i have no problem either way. 

> The only argument against is that we might be computing extra information...

I think its more likely that the user would want the center _and_ the radius rather than just one of them